pkg/backup/actions/csi/pvc_action.go

```diff
@@ -570,3 +581,485 @@ func NewPvcBackupItemAction(f client.Factory) plugincommon.HandlerInitializer {
 		}, nil
 	}
 }
+
+func (p *pvcBackupItemAction) getVolumeSnapshotReference(
+	ctx context.Context,
+	pvc corev1api.PersistentVolumeClaim,
+	backup *velerov1api.Backup,
+) (*snapshotv1api.VolumeSnapshot, error) {
+	vgsLabelKey := backup.Spec.VolumeGroupSnapshotLabelKey
+	group, hasLabel := pvc.Labels[vgsLabelKey]
+
+	if vgsLabelKey != "" && hasLabel && group != "" {
+		p.log.Infof("PVC %s/%s is part of VGS group %q via label %q", pvc.Namespace, pvc.Name, group, vgsLabelKey)
+
+		// Try to find an existing VS created via a previous VGS in the current backup
+		existingVS, err := p.findExistingVSForBackup(backup.UID, backup.Name, pvc.Name, pvc.Namespace)
+		if err != nil {
+			return nil, err
+		}
+
+		if existingVS != nil {
+			if existingVS.Status != nil && existingVS.Status.VolumeGroupSnapshotName != nil {
+				p.log.Infof("Reusing existing VolumeSnapshot %s for PVC %s", existingVS.Name, pvc.Name)
+				return existingVS, nil
+			} else {
+				return nil, errors.Errorf("found VolumeSnapshot %s for PVC %s, but it was not created via VGS (missing volumeGroupSnapshotName)", existingVS.Name, pvc.Name)
+			}
+		}
+
+		p.log.Infof("No existing VS found for PVC %s; creating new VGS", pvc.Name)
+
+		// List all PVCs in the VGS group
+		groupedPVCs, err := p.listGroupedPVCs(pvc.Namespace, vgsLabelKey, group)
+		if err != nil {
+			return nil, err
+		}
+
+		// Determine the CSI driver for the grouped PVCs
+		driver, err := p.determineCSIDriver(groupedPVCs)
+		if err != nil {
+			return nil, err
+		}
+		if driver == "" {
```

Comment:
The logic still could get an empty driver here, but the reason cannot be that multiple CSI drivers are found for the group.
The multiple drivers case is treated as an error in the function `determineCSIDriver`.
Please update the log here to make it clearer, e.g., `no driver found, fail the backup`.